src/main/java/com/example/gradecalculator/controller/UserController.java

```diff
@@ -76,6 +90,22 @@ public String userGet(Model model) {
         return "user";
     }
 
+    @PostMapping("/editProfile")
+    private String editProfilePost(@Valid @ModelAttribute UserEditTO editProfile, Authentication authentication){
+
+        var userService = (UserDetailsImpl)authentication.getPrincipal();
+        var userResult = userRepository.findById(userService.getId());
+        if(userResult.isEmpty()){
+            throw new IllegalArgumentException("User not found");
+        }
+        var user = userResult.get();
+        user.setFirstName(editProfile.getFirstName());
+        user.setLastName(editProfile.getLastName());
+        userRepository.save(user);
```

Comment:
As we start to have more complex functions in our controllers, I would say we should move "buisness logic" into a dedicated service. <br>
And luckly we already have such a class: `UserService`<br>
UserService should:
- recieve the TO
- do the mapping
- make repository calls

That reduces the noise in the controller class and enables better reuasability of logic methods

Reply:
Yes please, everything that you think can be moved to the service